js/background.js

```diff
@@ -416,7 +416,7 @@
         });
         var error = c.validateNumber();
         if (error) {
-            console.log('Invalid contact received', error && error.stack ? error.stack : error);
+            console.log('Invalid contact received:', Errors.toLogFormat(error));
```

Comment:
I just had a look at the `ensure-error` code, and it doesn't have the same behavior as our previous approach. Before, we looked for `.stack` and if we didn't find it, we would rely on the `toString()` of whatever we got. `ensure-error` does a `util.inspect()` on the incoming thing and creates a new callstack.

Seems like in most situations we'd be alright, perhaps even better than we were before. At least in terms of debuggability. But in some cases we'd show a whole lot more data than we did before since `util.inspect()` will enumerate keys while `toString()` didn't. That might have privacy implications.

Reply:
Thanks for digging in. I saw the behavior slightly changed but didn’t notice any exposure so far. Let’s keep an eye on it during the beta.